internal/fetch_config.go

```diff
@@ -6,32 +6,38 @@ import (
 )
 
 func NewFetchConfig(backupName,dbDataDirectory string, folder storage.Folder, spec *TablespaceSpec,
-	filesToUnwrap map[string]bool) *FetchConfig {
+	filesToUnwrap map[string]bool, skipRedundantTars bool) *FetchConfig {
 	fetchConfig := &FetchConfig{
-		filesToUnwrap:          filesToUnwrap,
-		missingBlocks:          make(map[string]int64),
-		tablespaceSpec:         spec,
-		backupName:             backupName,
-		folder:                 folder,
-		dbDataDirectory:        dbDataDirectory,
+		filesToUnwrap:     filesToUnwrap,
+		missingBlocks:     make(map[string]int64),
+		tablespaceSpec:    spec,
+		backupName:        backupName,
+		folder:            folder,
+		dbDataDirectory:   dbDataDirectory,
+		skipRedundantTars: skipRedundantTars,
 	}
 	return fetchConfig
 }
 
 type FetchConfig struct {
 	filesToUnwrap   map[string]bool
 	// missingBlocks stores count of blocks missing for file path
-	missingBlocks          map[string]int64
-	tablespaceSpec         *TablespaceSpec
-	backupName             string
-	folder                 storage.Folder
-	dbDataDirectory        string
+	missingBlocks     map[string]int64
+	tablespaceSpec    *TablespaceSpec
+	backupName        string
+	folder            storage.Folder
+	dbDataDirectory   string
+	skipRedundantTars bool
 }
 
 func (fc *FetchConfig) UpdateFetchConfig(unwrapResult *UnwrapResult) {
```

Comment:
Why is it called UpdateFetchConfig? I see the logic for skip redundant files.

Reply:
Fixed in https://github.com/wal-g/wal-g/pull/687/commits/0b2c8f6d2e118d0ed0d05ef1c711d5ba93ea089f